m_tunnel_key.go

```diff
@@ -0,0 +1,146 @@
+package tc
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/mdlayher/netlink"
+)
+
+const (
+	tcaTunnelUnspec = iota
+	tcaTunnelKeyTm
+	tcaTunnelKeyParms
+	tcaTunnelKeyEncIPv4Src
+	tcaTunnelKeyEncIPv4Dst
+	tcaTunnelKeyEncIPv6Src
+	tcaTunnelKeyEncIPv6Dst
+	tcaTunnelKeyEncKeyID
+	tcaTunnelKeyPad
+	tcaTunnelKeyEncDstPort
+	tcaTunnelKeyNoCSUM
+	tcaTunnelKeyEncOpts
+	tcaTunnelKeyEncTOS
+	tcaTunnelKeyEncTTL
+)
+
+// TunnelKey contains attribute of the TunnelKey discipline
+type TunnelKey struct {
+	Parms           *TunnelParms
+	Tm              *Tcft
+	KeyEncSrc       *net.IP
+	KeyEncDst       *net.IP	
+	KeyEncKeyID     *uint32
+	KeyEncDstPort   *uint16
+	KeyNoCSUM       *uint8
+	KeyEncTOS       *uint8
+	KeyEncTTL       *uint8
+}
+
+// TunnelParms from from include/uapi/linux/tc_act/tc_tunnel_key.h
+type TunnelParms struct {
+	Index           uint32
+	Capab           uint32
+	Action          uint32
+	RefCnt          uint32
+	BindCnt         uint32
+	TunnelKeyAction uint32
+}
+
+// marshalTunnelKey returns the binary encoding of TunnelKey
+func marshalTunnelKey(info *TunnelKey) ([]byte, error) {
+	options := []tcOption{}
+	if info == nil {
+		return []byte{}, fmt.Errorf("TunnelKey: %w", ErrNoArg)
+	}
+
+	if info.Parms != nil {
+		data, err := marshalStruct(info.Parms)
+		if err != nil {
+			return []byte{}, err
+		}
+		options = append(options, tcOption{Interpretation: vtBytes, Type: tcaTunnelKeyParms, Data: data})
+	}
+	if info.KeyEncSrc != nil {
```

Comment:
`info.KeyEncSrc` can either be IPv6 or IPv4. Using always `tcaTunnelKeyEncIPv4Src` might lead to interesting side effects.
And same for `...Dst`.

Reply:
I'll fix these issues - was doing some testing.

Reply:
Should be fixed now.